nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -392,22 +482,30 @@ def diarize(self, paths2audio_files: List[str] = None, batch_size: int = 0):
 
         self.AUDIO_RTTM_MAP = audio_rttm_map(self._diarizer_params.manifest_filepath)
 
+        out_rttm_dir = os.path.join(self._out_dir, 'pred_rttms')
+        os.makedirs(out_rttm_dir, exist_ok=True)
+
         # Speech Activity Detection
         self._perform_speech_activity_detection()
 
         # Segmentation
-        self._run_segmentation()
+        for scale_idx, (time_length, shift_length) in self.multiscale_args_dict['scale_dict'].items():
```

Comment:
I would suggest grouping all run_segmentation for [w1,w2,w3] to a single function so it can deal with window_len_in_sec and shift_len_in_sec directly and we can make subsegmentation and _extract_embeddings part more independent.

Reply:
Nithin and I discussed about this and we also wanted as you said, but unfortunately it can't be done with the current pipeline structure since we need to change the collate function each time we extract the embedding.

Reply:
make sense